spec/rspec/support/object_inspector_spec.rb

```diff
@@ -92,6 +92,17 @@ def with_date_loaded
             matcher_without_a_description.inspect)
         end
       end
+
+      context "with double objects" do
+        it "displays double name" do
+          double = double('Foo')
+          allow(double).to receive(:name).and_return("Foo")
+          ObjectInspector.register double.class do |double|
+            "Double (#{double.name})"
+          end
+          expect(ObjectInspector.inspect(double)).to eq("Double (Foo)")
+        end
+      end
```

Comment:
Rather than writing a spec for the exact case that's driving this (rspec-mocks test doubles) I think it's better to register a custom class and use that, for a few reasons:

* It proves that the register API works for _any_ arbitrary type, not just built-in rspec types.
* The registry is global state, which means that if you write a test here that registers inspect output for an `RSpec::Mocks::Double`, it will stomp the one that rspec-mocks will be registering once you work up that PR.
* What you've registered here works for this specific double (since you've stubbed `name`) but doesn't work for other doubles (which don't have `name` doubles).

So I'd recommend creating a custom type like `Currency` or `Person` and registering it, and using it in this spec.

We also want a spec showing that subclasses can use the registered inspect logic.